test/auto_parallel/pir/auto_parallel_recompute_pir_pass_unittest.py

```diff
@@ -0,0 +1,239 @@
+            batch_size=2,
+            shuffle=True,
+            drop_last=True,
+        )
+        train_dataloader = DataLoader(
+            train_dataset,
+            batch_sampler=train_sampler,
+            num_workers=0,
+        )
+        dist_loader = dist.shard_dataloader(
+            dataloader=train_dataloader,
+            meshes=[get_mesh(0), get_mesh(1)],
+            shard_dims="dp",
+        )
+        return optimizer, dist_loader
+
+    def run_llama(self, model_config):
+        self.init_dist_env()
+        # model
+        model = LlamaForCausalLMAuto(model_config)
+        criterion = LlamaPretrainingCriterionAuto(model_config)
+
+        optimizer, dist_loader = self.prepare_llama(model, model_config)
+
+        model = dist.to_static(
+            model, dist_loader, criterion, optimizer, strategy=self.strategy
+        )
+        model.train()
+
+        md5_losses = []
+        for step, inputs in enumerate(dist_loader()):
+            if step >= self.run_step:
+                break
+            input_ids, labels = inputs
+            loss = model(input_ids, labels)
+            array_bytes = np.array(loss).tobytes()
+            md5_losses.append(hashlib.md5(array_bytes).hexdigest())
+        return md5_losses, model
+
+    def init_dist_env(self):
+        order = ["dp", "pp", "mp"]
+        dp_degree = self.dp
+        mp_degree = self.mp
+        pp_degree = self.pp
+        degree = [dp_degree, pp_degree, mp_degree]
+        mesh_dims = list(filter(lambda x: x[1] > 1, list(zip(order, degree))))
+        if not mesh_dims:
+            mesh_dims = [("dp", 1)]
+        dim_names = [mesh_dim[0] for mesh_dim in mesh_dims]
+        mesh_shape = [mesh_dim[1] for mesh_dim in mesh_dims]
+        mesh_arr = np.arange(
+            0, reduce(lambda x, y: x * y, mesh_shape, 1)
+        ).reshape(mesh_shape)
+        global_mesh = dist.ProcessMesh(mesh_arr, dim_names)
+        dist.auto_parallel.set_mesh(global_mesh)
+        paddle.seed(1024)
+        np.random.seed(1024)
+        random.seed(1024)
+
+    def check_loss(self, losses_1, losses_2):
+        np.testing.assert_equal(len(losses_1), len(losses_2))
+        for idx in range(len(losses_1)):
+            np.testing.assert_equal(losses_1[idx], losses_2[idx])
+
+    def get_recompute_message(self, program):
+        segment_num = -1
+        rc_op_num = 0
+        for block in program.blocks:
+            for op in block.ops:
+                if op.has_attr("fwd_recompute_id"):
+                    idx = op.attrs()["fwd_recompute_id"]
+                    segment_num = max(segment_num, idx)
+                    rc_op_num += 1
+                elif op.has_attr("bwd_recompute_id"):
+                    idx = op.attrs()["bwd_recompute_id"]
+                    segment_num = max(segment_num, idx)
+                    rc_op_num += 1
+        return segment_num + 1, rc_op_num
+
+    def run_test_cases(self):
+        self.strategy._recompute.enable = False
```

Comment:
why `strategy.recompute` used in pass but `strategy._recompute` used here ?

Reply:
`strategy.recompute` (in pass) is the `recompute` attribute of class `paddle.distributed.Strategy()`
`strategy._recompute`(in engine) is the `_recompute` attribute of class `paddle.distributed.fleet.auto.Strategy()`
The two classes all describe the configuration information of `recompute` and can be converted in `to_static` funxtion